src/java/arjdbc/mysql/MySQLRubyJdbcConnection.java

```diff
@@ -247,7 +247,9 @@ protected Connection newConnection() throws RaiseException, SQLException {
             connection = super.newConnection();
         }
         catch (SQLException ex) {
-            if (ex.getErrorCode() == 1049) throw newNoDatabaseError(ex);
+            int errorCode = ex.getErrorCode();
+            // access denied, no database
+            if (errorCode == 1044 || errorCode == 1049) throw newNoDatabaseError(ex);
```

Comment:
access denied as well, hmm AR really keeps being opinionated :wink:

Reply:
yeah, but the thing is, the MySQL error is kinda wrong. What is tested and I tested manually was a missing database...and you get "access denied"...maybe if you try to access a non-existing DB with root access you get the "no database" error...dunno...